finalitygadget/interface.go

```diff
@@ -3,7 +3,7 @@ package finalitygadget
 import "github.com/babylonlabs-io/finality-gadget/types"
 
 type IFinalityGadget interface {
-	/* QueryIsBlockBabylonFinalized checks if the given L2 block is finalized by the Babylon finality gadget
+	/* QueryIsBlockBabylonFinalizedFromBabylon checks if the given L2 block is finalized by the Babylon finality gadget
```

Comment:
what's the difference w the old name?

Reply:
ok it seem `QueryIsBlockBabylonFinalizedFromBabylon` calculates quorum on the fly while `QueryBlockRangeBabylonFinalized` read from db

so I think for all external RPCs calls, we should always read from DB to avoid DoS attack.

in that case, should we remove the RPC?

Reply:
we can make `queryIsBlockBabylonFinalizedFromBabylon` internal to prevent unwanted calls. i left it external for now as we may need it for testing (e.g. if FG returns block not finalized, we would want to query Babylon node directly to debug)

Reply:
sure we can make it external for now for development. maybe add a TODO there so remove it in the future once everything is stable?